authentication/typescript/azure/src/azureIdentityAccessTokenProvider.ts

```diff
@@ -1,22 +1,35 @@
 import { GetTokenOptions, TokenCredential } from "@azure/core-auth";
-import { AccessTokenProvider } from '@microsoft/kiota-abstractions';
+import { AccessTokenProvider, AllowedHostsValidator } from '@microsoft/kiota-abstractions';
 
 export class AzureIdentityAccessTokenProvider implements AccessTokenProvider {
     /**
      *@constructor
      *@param credentials The tokenCredential implementation to use for authentication.
      *@param scopes The scopes to use for authentication.
      *@param options The options to use for authentication.
+     *@param allowedHosts The allowed hosts to use for authentication.
      */
-    public constructor(private readonly credentials: TokenCredential, private readonly scopes: string[] = ['https://graph.microsoft.com/.default'], private readonly options?: GetTokenOptions) {
+    public constructor(private readonly credentials: TokenCredential, 
+        private readonly scopes: string[] = ['https://graph.microsoft.com/.default'],
+        allowedHosts: string[] = ['graph.microsoft.com', 'graph.microsoft.us', 'dod-graph.microsoft.us', 'graph.microsoft.de', 'microsoftgraph.chinacloudapi.cn', 'canary.graph.microsoft.com'],
+        private readonly options?: GetTokenOptions) {
         if (!credentials) {
             throw new Error('parameter credentials cannot be null');
         }
         if (!scopes || scopes.length === 0) {
             throw new Error('scopes cannot be null or empty');
         }
+        this.allowedHostsValidator = allowedHosts ? new AllowedHostsValidator(...allowedHosts) : new AllowedHostsValidator();
     }
-    public getAuthorizationToken = async (): Promise<string> => {
+    private readonly allowedHostsValidator: AllowedHostsValidator;
+    public getAuthorizationToken = async (url?:string): Promise<string> => {
+        if(!url || !this.allowedHostsValidator.isUrlHostValid(url)) {
+            return '';
+        }
+        // @ts-ignore
+        if(!url.startsWith('https://') || (window && (window.location.protocol as string).toLowerCase() !== 'https:')) {
```

Comment:
This should part of host validation function and not in this file.

Reply:
Single responsibility principle, the host validator validates an URL is in the host names allow list.
Now, I get the argument for reusability. We could add a second method like `ValidateHttps` in that class, but then, what should the class name be?

Reply:
thoughts?

Reply:
I think a single util function should help us achieve this.

Reply:
moved to it's own function